schema/tfaction-root.json

```diff
@@ -398,12 +401,19 @@
         },
         "envs": {
           "$ref": "#/$defs/Envs"
+        },
+        "runs_on": {
+          "$ref": "#/$defs/RunsOn"
         }
       }
     },
     "Envs": {
       "type": "object",
       "description": "environment variables"
+    },
+    "RunsOn": {
+      "type": ["string", "array"],
```

Comment:
I'm not familiar with JSON Schema, so I'm not sure if this is correct.
I'll look into.

Reply:
https://cswr.github.io/JsonSchema/spec/multiple_types/
It seems correct, no ?

Reply:
📝 `type` can accept an array.

https://json-schema.org/draft/2020-12/json-schema-core

```
"type": ["object", "boolean"],
```

But I think `array` is ambiguous because the type of elements is unknown. 🤔

Reply:
Maybe this is better :
```
"RunsOn": {
  "oneOf": [
    {
      "type": "string"
    },
    {
      "type": "array",
      "items": {
        "type": "string"
      }
    },
    {
      "type": "null"
    }
  ],
  "description": "The type of runner that the job will run on"
}
```

Reply:
https://json-schema.org/understanding-json-schema/reference/combining#anyOf

Should we use `anyOf`?

```json
    "RunsOn": {
      "anyOf": [
        {
          "type": "string"
        },
        {
          "type": "array",
          "items": {
            "type": "string"
          }
        }
      ],
      "description": "The type of runner that the job will run on"
    }
```

Reply:
After reading more, Anyof seems more appropriate. It's more like a OR. So yes definitely.

Reply:
But null type is necessary because we can let empty the property no?

Reply:
If `runs_on` is unset, the type definition is unrelated. So I don't think we need to add `null`.
If we want to support the setting which sets `null` explictly, we need to add `null`.
But I don't think we need to support `null` explicitly.

```yaml
runs_on: null
```

Reply:
Fixed. 7290811a4c4481dc2f16f68d19e5cb4f8f85e890